Mathlib/Order/KrullDimension.lean

```diff
@@ -81,9 +81,27 @@ variable {α β : Type*}
 
 variable [Preorder α] [Preorder β]
 
+/--
+Alternative definition of height, with the supremum only ranging over those series
+that end at `a`.
+-/
+lemma height_eq_iSup_last_eq (a : α) :
+    height a = ⨆ (p : LTSeries α) (_ : p.last = a), ↑(p.length) := by
+  apply le_antisymm
+  · apply iSup₂_le
+    intro p hlast
+    -- The proof is a bit more elaborate since we have a Preorder, not a PartialOrder
+    wlog hlenpos :  p.length ≠ 0
+    · simp_all
+    let p' := p.eraseLast.snoc a ( lt_of_lt_of_le (p.eraseLast_last_rel_last hlenpos) hlast)
+    apply le_iSup₂_of_le p' (by simp [p']) (by simp [p']; norm_cast; omega)
+  · apply iSup₂_le
+    intro p hlast
+    apply le_iSup₂_of_le p (by simp [hlast]) (by simp)
+
 lemma height_le {x : α} {n : ℕ∞} :
-    height x ≤ n ↔ ∀ (p : LTSeries α), p.last ≤ x → p.length ≤ n := by
-  simp [height, iSup_le_iff]
+    (∀ (p : LTSeries α), p.last = x → p.length ≤ n) → height x ≤ n := by
+  simp [height_eq_iSup_last_eq, iSup_le_iff.mpr]
```

Comment:
Accident?

Reply:
No, see the PR description: As I was updating the development I found that the `height_le(_iff)` lemma is simply not needed anywhere, but I do often need to prove `height x ≤ n` and the stronger assumption (with the equality) is helpful.

I guess I could phrase this as an `iff` lemma, and use with `.mpr`, but what’s the point of an iff lemma if you only ever use one direction (for the other direction, it’s usually better to have the inquality, as that’s more general.

Reply:
> I guess I could phrase this as an `iff` lemma, and use with `.mpr`, but what’s the point of an iff lemma if you only ever use one direction (for the other direction, it’s usually better to have the inquality, as that’s more general.

It's the universal property of `height`. That seems rather important!

Whenever I define some kind of supremum/infimum, the universal property is the single most useful lemma in all of its API.

Reply:
Yeah, I had them originally. Will put them back. And I guess I'll just include both variants (with = and with ≤), maybe more efficient than trying to guess which ones will actually be needed.